internal/discord/commands.go

```diff
@@ -112,7 +112,73 @@ func (d *Discord) Purge(s *discordgo.Session, i *discordgo.InteractionCreate) {
 //	duration:	string
 //	reason:		string
 func (d *Discord) Exile(s *discordgo.Session, i *discordgo.InteractionCreate) {
-	return
+	optionMap := mapOptions(i)
+	logMsg, _ := d.LogCommand(i.Interaction)
+
+	state := &InteractionState{
+		session:     s,
+		interaction: i,
+		logMsg:      logMsg,
+		isFirst:     true,
+	}
+
+	// Calculate duration of exile
+	startTime := time.Now()
+	duration, err := parseDuration(optionMap["duration"].StringValue())
+	if err != nil {
+		return
+	}
+
+	userToExile := optionMap["user"].UserValue(nil)
+
+	err = d.ExileUser(state, userToExile.ID, optionMap["reason"].StringValue())
+	if err != nil {
+		return
+	}
+
+	// Inform invoker and edit log message of successful exile
+	endTime := startTime.Add(duration)
+	tempstr := fmt.Sprintf(
+		"User <@%v> has been exiled until <t:%v>",
+		userToExile.ID,
+		endTime.Unix(),
+	)
+	RespondAndAppendLog(state, tempstr)
+
+	// DM user regarding the exile, doesn't matter if DM fails
+	tempstr = fmt.Sprintf("You are being exiled from `%v` until <t:%v> for the following reason:\n> %v",
+		GuildName,
+		endTime.Unix(),
+		optionMap["reason"].StringValue(),
+	)
+	d.SendDMToUser(state, userToExile.ID, tempstr)
+	d.EditLogMsg(logMsg)
+
+	time.Sleep(duration)
```

Comment:
since the reqs changed, let's not even bother calculating the time if theres no duration arg to do so. that said we would probably need to have another tempstr case where there is no duration arg